---
title: "Longitudinal data exploratory analysis"
author: "Antonia Chroni <achroni@stjude.org> for St. Jude Children's Research Hospital BioHackathon Team 1"
output:
  html_document:
    toc: TRUE
    toc_float: TRUE
    df_print: paged
    code_folding: hide
    toc_depth: 6
    highlight: tango
    number_sections: TRUE
  pdf_document:
    toc: TRUE
    highlight: tango
    number_sections: TRUE
    latex_engine: lualatex
    keep_tex: FALSE
always_allow_html: TRUE
urlcolor: blue
linkcolor: black
citecolor: blue
geometry: margin=1in
header-includes: 
  - \usepackage{titling}
  - \usepackage{fancyhdr}
params:
  root_dir: './'
  PROJECT_NAME: './'
  LEAD_ANALYSTS: './'
  CONTACT_EMAIL: './'
  START_DATE: './'
  COMPLETION_DATE: './'
  input_file: './'
---

```{r logo-file, echo=FALSE}
attach(params)
# Insert logo on the top of the html report 
logo_file <- file.path(root_dir, "figures", "img", "chromic-logo.png")
htmltools::img(src = knitr::image_uri(logo_file), alt = "logo", style = "position:absolute; top:0; left:0; padding:0px; height:120px;")
detach(params)
```

\addtolength{\headheight}{2.0cm} 
\fancypagestyle{plain}{} 
\thispagestyle{fancy}
\fancyhead[L]{\includegraphics[height=120px]{`r logo_file`}}
\renewcommand{\headrulewidth}{0pt}

<style type="text/css">
:root { --chromic_color: #17818F;}

h1.title {margin-top: 130px;
  margin-bottom: 25px;
  font-size: 36px;}

.nobullet li {list-style-type: none;}

.reporthead {font-size: 20px;}

body { /* Normal */
  font-size: 16px;
  font-style: Arial, Helvetica, sans-serif;}

h1 {
  color: var(--chromic_color);
  font-size: 28px;
  margin-top: 50px;}

h2 {color: var(--chromic_color);
  font-size: 20px;}

.list-group-item.active, .list-group-item.active:focus, .list-group-item.active:hover {
  background-color: var(--chromic_color);}
</style>

<a href="https://wiki.stjude.org/display/CAB">

</a>

\pagebreak

<div class="reporthead"><br/>
**Project: `r params$PROJECT_NAME`**  

<br />  

**St. Jude Children's Research Hospital BioHackathon Team 1**
<br />  


>Date started: `r params$START_DATE`
<br />
>Date completed:  `r params$COMPLETION_DATE`
<br />
>Report generated: `r format(Sys.time(), '%H:%M:%S %Z %m/%d/%Y')` \

</div>
\pagebreak

# Information about this notebook
This is an exploratory analysis of the data availability in terms of assays in the Comprehensive Omics Catalogue for Hartwell. This is critical for mitigating duplicate sequencing requests and efforts through Hartwell. This notebook aims to showcase:
(1) which samples have already been sequenced by Hartwell, and 
(2) what omics data are available per sample.

For demo purposes, we use dummy data cohort and subset by human brain tumor samples. Finally, we investigate the number of samples per `cancer_type_brain`, `Assay` and `longitudinal`.

# Set up

```{r load-library, echo=TRUE}
suppressPackageStartupMessages({
  library(tidyverse)
})
```

# Directories and paths to file Inputs/Outputs

```{r set-dir-and-file-names, echo=TRUE}
attach(params)
analysis_dir <- file.path(root_dir, "analyses", "data-exploratory-analysis") 
input_dir <- file.path(analysis_dir, "input") 

# We will first read in metadata file as we need to define sample_name 
metadata_file <- file.path(input_dir, input_file) # metadata input file
palette_file <- file.path(root_dir, "figures", "palettes", "assay_color_palette.tsv")
tumor_palette_file <- file.path(root_dir, "figures", "palettes", "tumor_color_palette.tsv")
longitudinal_palette_file <- file.path(root_dir, "figures", "palettes", "longitudinal_color_palette.tsv")

# File path to `plots` directory
plots_dir <- file.path(analysis_dir, "plots") 

figures_plots_dir <- file.path(plots_dir, "figures-longitudinal") 
if (!dir.exists(figures_plots_dir)) {
  dir.create(figures_plots_dir)}

source(paste0(analysis_dir, "/util/function-create-barplot.R"))
source(paste0(root_dir, "/figures/scripts/theme_plot.R"))
```

# Read metadata file

We will subset by human brain tumor samples.

```{r read-metadata, echo=TRUE}
# Read metadata
df <- read.csv(metadata_file, stringsAsFactors=FALSE) 

# Number of samples per cancer_type_brain
assays_number <- length(df$SJUID)
samples_number <- length(unique(df$SJUID))
```

```{r re-order-df, echo=TRUE}
cancer_type_brain_order <- c("Ependymoma", "HGG", "LGG", "Medulloblastoma")
longitudinal_order <- c("Diagnosis", "Remission", "Relapse", "Deceased")

# Re-order df
f <- c("WES", "WGBS", "WGS", "RNAseq", "ATACseq", "ChIPseq", "Methylation")# Level df by assay
df <- df %>% 
  dplyr::mutate(Assay = factor(Assay),
                Assay = fct_relevel(Assay, f),
                longitudinal = factor(longitudinal),
                longitudinal = fct_relevel(longitudinal, longitudinal_order)) %>%
  arrange(cancer_type_brain, longitudinal, Assay) 
```

## Color palette for plotting

```{r palette, echo=TRUE}
# Read color palette
palette_df <- readr::read_tsv(palette_file, guess_max = 100000, show_col_types = FALSE)

# Define and order palette
palette <- palette_df$hex_codes
names(palette) <- palette_df$color_names

# Read color palette for tumor type
tumor_palette_df <- readr::read_tsv(tumor_palette_file, guess_max = 100000, show_col_types = FALSE)

# Define and order palette
tumor_palette <- tumor_palette_df$hex_codes
names(tumor_palette) <- tumor_palette_df$color_names


# Read color palette for longitudinal
longitudinal_palette_df <- readr::read_tsv(longitudinal_palette_file, guess_max = 100000, show_col_types = FALSE)

# Define and order palette
longitudinal_palette <- longitudinal_palette_df$hex_codes
names(longitudinal_palette) <- longitudinal_palette_df$color_names
```

# Number of samples with assay information

```{r, fig.align = "left", results = "asis", message = FALSE, warning = FALSE, echo = FALSE}
tables1 <- df %>% count(Assay)
cat("  \n<div style=\"font-size:80%\">  \n")
caption_value = "Summary of samples per assay"

print(knitr::kable(tables1, align = "lcccc", caption = caption_value))
cat("  \n</div>  \n")
cat("  \n\\pagebreak  \n")
```

```{r plot1, fig.width = 6, fig.height = 5, fig.fullwidth = TRUE}
# Define parameters for function
ylim <- max(tables1$n)

# Run function
fname <- paste0(figures_plots_dir, "/", "samples-per-assay-longitudinal.pdf")
p <- create_barplot(plot_df = tables1, 
                    ylim = ylim, 
                    x_value = tables1$Assay, 
                    use_palette = palette, 
                    xtitle ="Type of assay", 
                    title_value = caption_value)
pdf(file = fname, width = 6, height = 5)
print(p)
dev.off()
```

# Number of samples per brain cancer type and assay

## Overall assays

There are `r samples_number` brain tumor samples with `r assays_number` assays in total.

```{r, fig.align = "left", results = "asis", message = FALSE, warning = FALSE, echo = FALSE}
tables1 <- df %>% count(cancer_type_brain)
cat("  \n<div style=\"font-size:80%\">  \n")
caption_value = "Summary of samples and assays per brain cancer type"

print(knitr::kable(tables1, align = "lcccc", caption = caption_value))
cat("  \n</div>  \n")
cat("  \n\\pagebreak  \n")
```

```{r plot2, fig.width = 6, fig.height = 5, fig.fullwidth = TRUE}
# Define parameters for function
ylim <- max(tables1$n)

# Run function
fname <- paste0(figures_plots_dir, "/", "cancer-type-brain-assay-longitudinal-overall.pdf")
p <- create_barplot(plot_df = tables1, 
                    ylim = ylim, 
                    x_value = tables1$cancer_type_brain, 
                    use_palette = tumor_palette, 
                    xtitle ="Brain tumor type",
                    title_value = caption_value)
pdf(file = fname, width = 6, height = 5)
print(p)
dev.off()
```

## Per assay

```{r, fig.align = "left", results = "asis", message = FALSE, warning = FALSE, echo = FALSE}
tables1 <- df %>% count(cancer_type_brain, Assay) %>%
  pivot_wider(names_from = Assay, values_from = n) %>%
  as.data.frame() %>%
  mutate_all(funs(replace_na(.,0)))

caption_value = "Summary of samples and assays per brain cancer type and per assay"

cat("  \n<div style=\"font-size:80%\">  \n")
print(knitr::kable(tables1, align = "lcccc", caption = caption_value))
cat("  \n</div>  \n")
cat("  \n\\pagebreak  \n")
```

```{r plot3, fig.width = 6, fig.height = 5, fig.fullwidth = TRUE}
tables1 <- df %>% count(cancer_type_brain, Assay) %>%
  as.data.frame() %>%
  mutate_all(funs(replace_na(.,0)))

# Plot stacked barplot 
fname <- paste0(figures_plots_dir, "/", "cancer-type-brain-per-assay-longitudinal.pdf")
p <- create_stacked_barplot(plot_df = tables1, 
                            x_value = tables1$cancer_type_brain, 
                            use_palette = palette, 
                            xtitle ="Brain tumor type", 
                            legend = tables1$Assay,
                            title_value = caption_value)
pdf(file = fname, width = 6, height = 5)
print(p)
dev.off() 
```

## Per disease stage

```{r, fig.align = "left", results = "asis", message = FALSE, warning = FALSE, echo = FALSE}
tables1 <- df %>% count(cancer_type_brain, longitudinal, Assay) %>%
  pivot_wider(names_from = Assay, values_from = n) %>%
  as.data.frame() %>%
  mutate_all(funs(replace_na(.,0)))

caption_value = "Summary of samples and assays per brain cancer type, per disease stage and per assay"

cat("  \n<div style=\"font-size:80%\">  \n")
print(knitr::kable(tables1, align = "lcccc", caption = caption_value))
cat("  \n</div>  \n")
cat("  \n\\pagebreak  \n")
```

# Number of samples per brain cancer type, longitudinal, assay, and SJUID

```{r, fig.align = "left", results = "asis", message = FALSE, warning = FALSE, echo = FALSE}
tables1 <- df %>% count(cancer_type_brain, SJUID, longitudinal, Assay) %>%
  pivot_wider(names_from = Assay, values_from = n) %>%
  as.data.frame() %>%
  mutate_all(funs(replace_na(.,0)))

cat("  \n<div style=\"font-size:80%\">  \n")
print(knitr::kable(tables1, align = "lcccc", caption = "Summary of samples and assays per brain cancer type, per disease stage, per assay and per SJUID"))
cat("  \n</div>  \n")
cat("  \n\\pagebreak  \n")
```

```{r plot4, fig.width = 6, fig.height = 5, fig.fullwidth = TRUE}
tables1 <- df %>% count(cancer_type_brain, SJUID, longitudinal, Assay) %>%
  as.data.frame() %>%
  mutate_all(funs(replace_na(.,0))) %>%
  
  # Add col
  #add_column(value = "1") %>%
  mutate(value = case_when(grepl("Diagnosis", longitudinal) ~ "1",
                           grepl("Remission", longitudinal) ~ "2",
                           grepl("Relapse", longitudinal) ~ "3",
                           grepl("Deceased", longitudinal) ~ "4")) %>%
  arrange(cancer_type_brain, longitudinal, Assay) 


# Dumbbell plot per cancer group
cancer_groups <- unique(as.character(tables1$cancer_type_brain))
cancer_groups <- sort(cancer_groups, decreasing = FALSE)
print(cancer_groups)

for (i in seq_along(cancer_groups)) {
  print(i)
  tables1_sub <- tables1 %>% 
    filter(cancer_type_brain == cancer_groups[i]) %>%
    arrange(cancer_type_brain, longitudinal, Assay) 
  
    # Name plots
   fname <- paste0(figures_plots_dir, "/", cancer_groups[i], "-cancer-type-brain-per-assay-longitudinal-disease stage.pdf")
    print(fname)
    
    # Run function
    p <- create_dumbbell_ct(plot_df = tables1_sub, 
                            x_value = tables1_sub$cancer_type_brain, 
                            use_palette = longitudinal_palette, 
                            #legend = tables1_sub$longitudinal,
                            ct_id = cancer_groups[i])
    pdf(file = fname, width = 12, height = 8)
    print(p)
    dev.off()
}







# Plot stacked barplot 
fname <- paste0(figures_plots_dir, "/", "cancer-type-brain-per-assay-longitudinal-disease stage.pdf")
p <- create_dumbbell_ct(plot_df = tables1, 
                            x_value = tables1$cancer_type_brain, 
                            use_palette = palette, 
                            legend = tables1$Assay,
                            title_value = caption_value)
pdf(file = fname, width = 6, height = 5)
print(p)
dev.off() 
```

# Future directions

The current exploratory data analysis module can be expanded by investigating samples with paired assays. Moreover, if other metadata are available, e.g., disease_stage, treatment, this will build large, longitudinal cohorts with multi-omic sequencing data. Such an analysis permits consideration of samples according to the condition(s) of the experiment and research aims. In addition, it can be used to refine research questions and/or generate new ones.

This will facilitate collaboration across departments at St. Jude, expedite discoveries, and find cures for children with cancer and other catastrophic diseases.

```{r echo=FALSE}
detach(params)
```

\pagebreak

# Session Info

```{r echo=FALSE}
sessionInfo()
```

